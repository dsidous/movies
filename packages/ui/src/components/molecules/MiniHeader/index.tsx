import React from 'react';
import { Link } from 'react-router-dom';
import { Box, Typography } from '@material-ui/core';

import { MediaImage } from '@atoms/index';
import useStyles from './styles';

interface Props {
  title: string;
  release_date: string;
  poster_path: string;
  link: string;
  linkCopy: string;
}

const MiniHeader: React.FC<Props> = ({
  title,
  release_date,
  poster_path,
  link,
  linkCopy,
}) => {
  const classes = useStyles();

  return (
    <div className={classes.root}>
      <Box className={classes.inner}>
        <MediaImage
          mediaType='poster'
          size={1}
          filePath={poster_path}
          name={title}
        />
        <Box className={classes.info}>
          <Typography component='div'>
            <Box fontWeight={700}>
              {title}
              &nbsp;
              {release_date && <span>({release_date.slice(0, 4)})</span>}
            </Box>
          </Typography>
          <Typography variant='body2'>
            <Link to={link}>
              &#8592;
              {linkCopy}
            </Link>
          </Typography>
        </Box>
      </Box>
    </div>
  );
};

export default MiniHeader;
