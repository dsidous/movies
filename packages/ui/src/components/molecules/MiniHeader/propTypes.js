import { string } from 'prop-types';

export const propTypes = {
  title: string.isRequired,
  release_date: string.isRequired,
  poster_path: string.isRequired,
  link: string.isRequired,
  linkCopy: string.isRequired,
};

export const defaultProps = {};
