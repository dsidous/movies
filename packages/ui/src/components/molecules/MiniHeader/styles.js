import { makeStyles } from '@material-ui/core/styles';

export default makeStyles(
  (theme) => ({
    root: {
      borderBottom: theme.separatorBold,
      padding: theme.spacing(4, 3),
    },

    inner: {
      display: 'grid',
      gridTemplateColumns: '70px 4fr',
      gridColumnGap: theme.spacing(2),
      height: theme.spacing(13),
      margin: '0 auto',

      '& img': {
        height: '100%',
        maxWidth: '70px',
      },
    },

    info: {
      alignSelf: 'center',
      display: 'flex',
      flexDirection: 'column',
      justifyItems: 'center',
    },
  }),
  { name: 'miniHeader' }
);
