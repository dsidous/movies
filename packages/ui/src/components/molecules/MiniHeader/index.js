import React from 'react';
import { Link } from 'react-router-dom';
import { Box, Typography } from '@material-ui/core';

import { propTypes } from './propTypes';
import MediaImage from '../../atoms/MediaImage';
import useStyles from './styles';

const MiniHeader = ({ title, release_date, poster_path, link, linkCopy }) => {
  const classes = useStyles();

  return (
    <div className={classes.root}>
      <Box className={classes.inner}>
        <MediaImage
          mediaType="poster"
          size={1}
          filePath={poster_path}
          name={title}
        />
        <Box className={classes.info}>
          <Typography component="div">
            <Box fontWeight={700}>
              {title}
              &nbsp;
              {release_date && <span>({release_date.slice(0, 4)})</span>}
            </Box>
          </Typography>
          <Typography variant="body2">
            <Link to={link}>
              &#8592;
              {linkCopy}
            </Link>
          </Typography>
        </Box>
      </Box>
    </div>
  );
};

MiniHeader.propTypes = propTypes;

export default MiniHeader;
