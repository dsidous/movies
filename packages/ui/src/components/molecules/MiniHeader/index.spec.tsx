import React from 'react';
import { shallow } from 'enzyme';
import MiniHeader from '.';

const mockProps = {
  title: 'title',
  release_date: '01.01.2010',
  poster_path: 'path/to/image',
  link: 'url',
  linkCopy: 'lnkCopy',
};

describe('Molecules/MiniHeader', () => {
  it('should render as expected', () => {
    const wrapper = shallow(<MiniHeader {...mockProps} />);

    expect(wrapper).toMatchSnapshot();
  });
});
