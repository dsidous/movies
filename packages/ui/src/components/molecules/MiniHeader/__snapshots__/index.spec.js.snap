// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Molecules/MiniHeader should render as expected 1`] = `ShallowWrapper {}`;
